### PR TITLE
rafthttp: add metrics for sending message

### PR DESCRIPTION
**Pull request by xiang90**
record message sending latency.

### CODE REVIEW

File: rafthttp/stream.go [hunk 1]
```diff
@@ -107,6 +108,7 @@ func (cw *streamWriter) run() {
 				continue
 			}
 			flusher.Flush()
+			reportSendingDuration(string(t), linkHeartbeatMessage, time.Since(start))
```
Comment (yichengq) [inline rafthttp/stream.go]: string(t) returns either `message` or `msgapp`, which is a little unclear.

Comment (xiang90) [inline rafthttp/stream.go]: we have three channel: message, msgapp and pipeline.

File: rafthttp/pipeline.go [hunk 2]
```diff
@@ -121,6 +121,7 @@ func (p *pipeline) handle() {
 			if isMsgSnap(m) {
 				p.r.ReportSnapshot(m.To, raft.SnapshotFinish)
 			}
+			reportSendingDuration(pipelineMsg, m, time.Since(start))
```
Comment (yichengq) [inline rafthttp/pipeline.go]: end.Sub(start) is more accurate for measurement

Comment (xiang90) [inline rafthttp/pipeline.go]: we actually do not care about the nano level latency. I am also going to remove the `end` when i start to remove the stats stuff.

Comment (yichengq) [inline rafthttp/pipeline.go]: yup. make sense.

**Comment by yichengq** [comment]
lgtm

**Pull request merged**